crates/sol-types/src/types/interface.rs

```diff
@@ -87,7 +90,7 @@ pub enum ContractError<T> {
     Panic(Panic),
 }
 
-impl<T: SolCalls> SolCalls for ContractError<T> {
+impl<T: SolInterface> SolInterface for ContractError<T> {
```

Comment:
this allows accidentally mixing calls and errors, yeah?

`ContractError<MyContractCalls>` is valid, as is `ContractError<MyContractErrors>`

is there a way to fix this in the type system?

Reply:
Would either require a generic or assoc type which implements `Sol*`, but idk.
I'd also like to not have different traits for the event enum with something like `type Selector = [u8; _]` but not sure if doable.
I'd leave this for later since it's not that big of a deal.

Reply:
should we have a trait to capture this somehow? 

```
trait SelectorPrefixed {}
trait SolError: SelectorPrefixed {}
```
that sort of thing?